packages/histoire/src/node/stories.ts

```diff
@@ -48,14 +48,20 @@ export async function watchStories(newContext: Context) {
   const watcher = chokidar.watch('.', {
     cwd: context.root,
     ignored: (path, stats) => {
-      if (context.config.storyIgnored.some(pattern => micromatch.isMatch(path, pattern))) {
+      // Normalize to relative path for pattern matching
+      const relativePath = resolve(path).startsWith(context.root)
+        ? relative(context.root, resolve(path))
+        : path
+
+      if (context.config.storyIgnored.some(pattern => micromatch.isMatch(relativePath, pattern))) {
         return true
       }
-      if (context.config.storyMatch.some(pattern => micromatch.isMatch(path, pattern))) {
+      if (context.config.storyMatch.some(pattern => micromatch.isMatch(relativePath, pattern))) {
         return false
       }
 
-      return stats?.isFile()
+      // Allow directories to be traversed, only ignore files that don't match
+      return stats?.isFile() ?? false
```

Comment:
I still don't understand this 🤔... Is this just for typescript inference?
Isn't something like this a bit more readable? `return !!stats?.isFile()`